src/tools/pkcs11-tool.c

```diff
@@ -2332,10 +2344,20 @@ static void sign_data(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 	if (sz < 0)
 		util_fatal("Cannot read from %s: %m", opt_input);
 
+	CK_MAC_GENERAL_PARAMS mac_gen_param;
 	if (opt_mechanism == CKM_RSA_PKCS_PSS && (size_t)sz != hashlen) {
 		util_fatal("For %s mechanism, message size (got %z bytes) "
 			"must be equal to specified digest length (%lu)\n",
 			p11_mechanism_to_name(opt_mechanism), sz, hashlen);
+	} else if (opt_mechanism == CKM_AES_CMAC_GENERAL) {
+		if (opt_mac_gen_param == 0 || opt_mac_gen_param > 16) {
+			util_fatal("For %s mechanism, the value CK_MAC_GENERAL_PARAMS"
+				"must be comprised between 1 and 16 (>=8 recommended).\n",
```

Comment:
in this error message, I would prefer to reference the cli option name used instead of the PKCS#11 value (or both) as this is what user provided and what was wrong. User reading this error needs to backtrack from the parameter name to the cli option, which might not be straightforward.